torch/lib/THS/generic/THSTensorMath.c

```diff
@@ -181,18 +419,38 @@ void THSTensor_(spcadd)(THTensor *r_, THTensor *dense, real value, THSTensor *sp
   THTensor      *values = THSTensor_(values)(sparse);
   THLongStorage *storage = THSTensor_(newSizeOf)(sparse);
   long          *sizes = storage->data;
+  long          nDim = THTensor_(nDimension)(dense);
+  long          nDimI = THSTensor_(nDimensionI)(sparse);
 
   THTensor_(resizeAs)(r_, dense);
   THSTensor_(contiguous)(sparse);
 
   if (r_ != dense) THTensor_(copy)(r_, dense);
 
-#pragma omp parallel for private(k)
-  for (k = 0; k < sparse->nnz; k++) {
-    long index = r_->storageOffset;
-    for (long d = 0; d < sparse->nDimension; d++)
-      index += r_->stride[d] * THTensor_fastGet2d(indices, d, k);
-    r_->storage->data[index]  += value * THTensor_fastGet1d(values, k);
+
+  if (nDim > nDimI) {
+    THTensor *srcBuffer = THTensor_(new)();
+    THTensor *dstBuffer = THTensor_(new)();
+    #pragma omp parallel for private(k)
+    for (k = 0; k < sparse->nnz; k++) {
```

Comment:
So far I've only tested with `OMP_NUM_THREADS=1` so I guess it doesn't work. I'll see if I can fix it (I'm not super familiar with OpenMP)